src/IrcConnection/IrcConnection.cs

```diff
@@ -1254,11 +923,12 @@ private void _Worker()
 #if LOG4NET
                         Logger.Socket.Warn("connection lost");
 #endif
+                        // CHECK: shouldn't be necessary any longer
                         // only flag this as connection error if we are not
                         // cleanly disconnecting
-                        if (!_Connection.IsDisconnecting) {
-                            _Connection.IsConnectionError = true;
-                        }
+                        //if (!_Connection.IsDisconnecting) {
+                        //    _Connection.IsConnectionError = true;
+                        //}
```

Comment:
Such changes can quickly lead the regressions, I think we will need test-case coverage, else nobody can tell what will happen.